scripts/truffle/bot.js

```diff
@@ -0,0 +1,304 @@
+const { BN } = require("@openzeppelin/test-helpers");
+const { sleep } = require("../../test-utils/utils");
+const { RAY, RAD, ONE_ETH } = require("../../test-utils/constants");
+
+const mcdJSON = require("../../config/mcd_testchain.json");
+const bpJSON = require("../../config/bprotocol_testchain.json");
+const { default: Web3 } = require("web3");
+
+// MCD Contracts
+const DssCdpManager = artifacts.require("DssCdpManager");
+const Dai = artifacts.require("Dai");
+const DaiJoin = artifacts.require("DaiJoin");
+const GemJoin = artifacts.require("GemJoin");
+const WETH9 = artifacts.require("WETH9");
+const Vat = artifacts.require("Vat");
+const Spotter = artifacts.require("Spotter");
+const LiquidatorInfo = artifacts.require("LiquidatorInfo");
+
+// B.Protocol Contracts
+const BCdpManager = artifacts.require("BCdpManager");
+const Pool = artifacts.require("Pool");
+const OSM = artifacts.require("OSM");
+
+let bCdpManager;
+let pool;
+let dssCdpManager;
+let gemJoin;
+let weth;
+let vat;
+
+let spot;
+let liqInfo;
+
+let MEMBER_1 = bpJSON.MEMBER_1;
+
+const ILK_ETH = web3.utils.padRight(web3.utils.asciiToHex("ETH-A"), 64);
+
+const topped = new Map();
+const bitten = new Map();
```

Comment:
it does not seem you reset it anywhere, and i am not sure we need it. also for topped.